src/condition/cmdi_detector.cpp

```diff
@@ -0,0 +1,263 @@
+    windows_command_opt,
+};
+
+// An iterator which returns the given scalar, so that the match_iterator can be
+// used directly with a scalar without the need for a fully-fledged object iterator
+class scalar_iterator {
+public:
+    explicit scalar_iterator(const ddwaf_object *obj, const std::span<const std::string> & /*path*/,
+        const exclusion::object_set_ref & /*exclude*/, const object_limits & /*limits*/)
+        : current_(obj)
+    {}
+
+    ~scalar_iterator() = default;
+
+    scalar_iterator(const scalar_iterator &) = default;
+    scalar_iterator(scalar_iterator &&) noexcept = default;
+
+    scalar_iterator &operator=(const scalar_iterator &) = delete;
+    scalar_iterator &operator=(scalar_iterator &&) noexcept = delete;
+
+    [[nodiscard]] const ddwaf_object *operator*() { return current_; }
+    bool operator++()
+    {
+        current_ = nullptr;
+        return false;
+    }
+    [[nodiscard]] explicit operator bool() const { return current_ != nullptr; }
+    [[nodiscard]] static std::vector<std::string> get_current_path() { return {}; }
+
+protected:
+    const ddwaf_object *current_;
+};
+
+// Most shells support -c as a way to specify a shell command, however some
+// shells such as ksh allow for the first argument to be a shell command
+std::unordered_map<std::string_view, std::string_view> known_shells{
+    {"sh", "c"},
+    {"bash", "c"},
+    {"ksh", {}},
+    {"rksh", {}},
+    {"fish", "c"},
+    {"zsh", "c"},
+    {"dash", "c"},
+    {"ash", "c"},
+    {"powershell", "Command"},
+    {"powershell.exe", "Command"},
+};
+
+std::string_view basename(std::string_view path)
+{
+    auto idx = path.find_last_of(R"(\/)"sv);
+    return idx == std::string_view::npos ? path : path.substr(idx + 1);
+}
+
+std::string_view trim_whitespaces(std::string_view str)
+{
+    static const std::string_view whitespaces = " \f\n\r\t\v";
+
+    if (str.empty()) {
+        return str;
+    }
+
+    auto start = str.find_first_not_of(whitespaces);
+    if (start == std::string_view::npos) {
+        return str;
+    }
+
+    auto end = str.find_last_not_of(whitespaces);
+    return str.substr(start, 1 + end - start);
+}
+
+std::size_t object_size(const ddwaf_object &obj) { return static_cast<std::size_t>(obj.nbEntries); }
+
+std::string_view object_at(const ddwaf_object &obj, std::size_t idx)
+{
+    const ddwaf_object &child = obj.array[idx];
+    if (child.type != DDWAF_OBJ_STRING) {
+        return {};
+    }
+    return {child.stringValue, object_size(child)};
+}
+
+std::string_view find_shell_command(std::string_view executable, const ddwaf_object &exec_args)
+{
+    // By initialising the cow_string with the underlying data
+    // contained within the std::string, we ensure a new one won't be
+    // allocated once the string is modified.
+    cow_string executable_lc{executable};
+    if (transformer::lowercase::transform(executable_lc)) {
+        executable = static_cast<std::string_view>(executable_lc);
+    }
```

Comment:
This transformation should be limited to windows.

Reply:
True, I need to split the windows / linux logic.